### ISSUE TITLE
[Errno 13] Permission denied: 'D:\\...\\vmlinuz'

### ISSUE THREAD

**Issue by Javapraca** [hakuna-m/wubiuefi#30]
I try to install Ubuntu on 32-bit Windows XP by:
- placing 32-bit desktop DVD ISO next to wubi.exe (in two ways - with network on and off),
- letting WUBI to download proper ISO,
but every time I get the following error (please find the logs as an attachment):
**ERROR  root: [Errno 13] Permission denied: 'D:\\ubuntu\\install\\boot\\vmlinuz' Traceback (most recent call last):**

Could you please help me with this?

Best regards

[wubi-16.10-rev317.log.txt](https://github.com/hakuna-m/wubiuefi/files/625914/wubi-16.10-rev317.log.txt)
[wubi-16.10-rev318.log.txt](https://github.com/hakuna-m/wubiuefi/files/625917/wubi-16.10-rev318.log.txt)

**Comment by hakuna-m** [comment]
wubi-16.10-rev317.log.txt and wubi-16.10-rev318.log.txt contain lines which indicates a previous install:

`11-22 16:15 DEBUG  WindowsBackend: previous_target_dir=D:\ubuntu`

There is nothing wrong with these lines because you tried the install again and again. But I miss the lines for deleting of the previous target directory: e.g.
 
`CommonBackend: Deleting D:\ubuntu

These lines are part of the log of the previous installer/uninstaller version, of course. But there should be at least a line:

`INFO   root: Running the uninstaller...`

If there is a previous uninstaller, there should be a line

`INFO   CommonBackend: Launching previous uninestaller D:\ubuntu\uninstall-wubi.exe `

If the version of the previous uninstaller is as equal as the current version, there should be a line:

`INFO   CommonBackend: This is the uninstaller running`

To put in a nutshell, can you check if the old target directory D:\ubuntu is completely removed before you install a new version.

Maybe, it is necessary to (uninstall it manually)[https://wiki.ubuntu.com/WubiGuide#How_do_I_manually_uninstall_Wubi.3F]

**Comment by Javapraca** [comment]
Yes, I forgot to mention that after each failed installation I removed the installation directory (D:\ubuntu) manually. I guess one time I allowed WUBI to delete previous installation. Anyway, I am sure that the installation directory did not exist when I run the installations (I think WUBI would not even let me leave its initial window, if the installation folder was not deleted).

I am also sure that WUBI created file D:\ubuntu\install\boot\vmlinuz .

Do you have any clues why WUBI has a problem accessing the file? Could this be WindowsXP failure?

**Comment by hakuna-m** [comment]
> Do you have any clues why WUBI has a problem accessing the file?

Wubi needs only read permissions for **D:\ubuntu\install\boot\vmlinuz**. IMHO if the file doesn't exist, there would be another error message. So you confirmed my assumption:
> I am also sure that WUBI created file D:\ubuntu\install\boot\vmlinuz

> I think WUBI would not even let me leave its initial window, if the installation folder was not deleted
In general, you are right. But maybe a hidden folder would be not deleted.

IMHO there are the following possible reasons:

 * wrong file attributes. e.g. the file is hidden
 * the sub task runs with the permissions of an underprivileged user
 * some Antivirus software blocks reading of the file for security reasons
 * the used Python version 2.7 has problems with Windows XP because XP is not supported by Microsoft anymore.

**Comment by Javapraca** [comment]
I have tried to install Ubuntu with WUBI on another Windows XP computer (also on D partition) and got the same errors. Both computers have installed Avast antivirus, but there were no warnings from it. The file attributes are set correctly.

Would you please try to replicate the error and fix it. I think that WUBI on Windows XP is most needed/demanded since XP users are practically forced to change their OS nowadays.

**Comment by hakuna-m** [comment]
> Would you please try to replicate the error

Some hours ago, I tried to replicate the error on an old computer with Windows XP SP3. The computer has no internet connection for security reasons. So I copied wubi1610r319.exe and a matching 32 bit ISO to my old computer and tried to install ....

But no error occured, the installation was successful.

**Comment by hakuna-m** [comment]
I compared my log and your logs and I found only minor differences.
 * I didn't use Internet connection, so the step "metalink md5sums comparing" was skipped.
 * Avast antivirus has never been installed on my old XP system. There is an old version of Avira Antivir which was installed some years ago when this system had its last internet connection.
 * I used wubi1610r319.exe instead of wubi1610r318.exe or wubi1610r317.exe but IMHO the differences between r317, r318 and r319 are not important for your issue.
 * I used **lubuntu-16.04.1-desktop-i386.iso** instead of **ubuntu-16.10-desktop-i386.iso**. This also skips some internet checks even if there is an internet connection.  
 * I used drive F: instead of drive D: But I don't think that it is important. Maybe there is a difference between system drive C: and other drives.

Here is the part of the log where I expected errors: 
```
02-10 14:46 DEBUG  CommonBackend: Extracting files from ISO F:\ubuntu\install\installation.iso
02-10 14:46 DEBUG  WindowsBackend:   extracting md5sum.txt from F:\ubuntu\install\installation.iso
02-10 14:46 DEBUG  WindowsBackend:   extracting casper\vmlinuz from F:\ubuntu\install\installation.iso
02-10 14:46 DEBUG  WindowsBackend:   extracting casper\initrd.lz from F:\ubuntu\install\installation.iso
02-10 14:46 DEBUG  CommonBackend: Checking kernel, initrd and md5sums
02-10 14:46 DEBUG  CommonBackend:   checking F:\ubuntu\install\boot\vmlinuz
02-10 14:46 DEBUG  CommonBackend:   F:\ubuntu\install\boot\vmlinuz md5 = 51e5984c776b1c420ced96f9f88b0116 == 51e5984c776b1c420ced96f9f88b0116
02-10 14:46 DEBUG  CommonBackend:   checking F:\ubuntu\install\boot\initrd.lz
02-10 14:46 DEBUG  CommonBackend:   F:\ubuntu\install\boot\initrd.lz md5 = 11bf9a6fae56bf7a9b426008033d5397 == 11bf9a6fae56bf7a9b426008033d5397
02-10 14:46 DEBUG  TaskList: ## Finished extract_kernel 
```

So maybe it works, if you disconnect internet during installation or if you use wubi1610r.... with a 16.04 iso.  This should skip some steps.
